model/project.go

```diff
@@ -1019,25 +1019,30 @@ func (p *Project) FindDistroNameForTask(t *task.Task) (string, error) {
 }
 
 func FindLastKnownGoodProject(identifier string) (*Project, error) {
+	const retryCount = 5
```

Comment:
I disagree with the retry approach. Retries make sense when an error is transient, like if you are querying a third-party service that might be briefly unavailable. In this case though the error is permanent: If someone commits a broken YAML, subsequent YAMLs will also presumably be broken. I think in cases like that it's better to error quickly.

Reply:
This isn't a retry - we're finding the next most recent version in case the most recent has unparseable yaml

Reply:
My claim is that eventually (5 commits after the bug is introduced) things will break. This only defers the time when the problem will become apparent.

Reply:
We can log a parsing error at a high level so that it's more visible. We shouldn't let a bug bring down the entire UI

Reply:
> We shouldn't let a bug bring down the entire UI

I 100% agree. My objection to this approach is that it doesn't accomplish that goal. I don't think we can rely on users to notice they've introduced a config error before N versions have gone by.

Reply:
For the most part this isn't anything with what a user can do, since introducing invalid yaml will be caught in the validator. This is for when we introduce a bug with saving/loading yaml. In that case we don't want the deploy to bring down the UI for a project

Reply:
I argue that we still want to break the UI in that case, since it's a much louder alarm, which will encourage us to revert faster. Keeping around code that may create bad data is very risky.

Reply:
Having users be the alarm isn't a good way to go about this. I'm all for logging something at critical level if we hit an error like this, but the UI should still work while we fix it

Reply:
Logging at critical SGTM. Is there a good place to do that?

Reply:
@ablack12 let's add a critical log statement at https://github.com/evergreen-ci/evergreen/pull/2940/files/6994696ec7ba4d44a037b48b238967e9adef67b3#diff-2baa3aff6c4b81bced2541140d1b2653R1040

Reply:
I'm glad I could help with this conversation 😂

Reply:
That's an `if err == nil` block, so I think you mean slightly later, but otherwise that SGTM.